model/dependencies.go

```diff
@@ -102,20 +108,39 @@ func (di *dependencyIncluder) expandDependencies(pair TVPair, depends []TaskUnit
 		case d.Variant == AllVariants && d.Name == AllDependencies: // task = *, variant = *
 			// Here we get all variants and tasks (excluding the current task)
 			// and add them to the list of tasks and variants.
-			for _, v := range di.Project.FindAllVariants() {
-				for _, t := range di.Project.FindTasksForVariant(v) {
-					if !(t == pair.TaskName && v == pair.Variant) {
-						deps = append(deps, TVPair{TaskName: t, Variant: v})
+			for _, v := range di.Project.BuildVariants {
+				for _, t := range v.Tasks {
+					projectTask := di.Project.FindTaskForVariant(t.Name, v.Name)
+					if projectTask != nil {
+						if projectTask.SkipOnRequester(di.requester) {
```

Comment:
can 114 - 121 be made a helper function since the logic is the same in 3 places? like 
```
if dep := dependencyForTask(projectTask, pair, requester); dep != nil {
  deps = append(deps, dep)
``` 
or something like that 🤷   i think making this modular will lower the risk of someone adding/changing just one of these cases on accident

Reply:
I actually tried that at first, but there are subtle differences in each that make having a helper function not be helpful. The code being repeated is some pretty simple stuff so I think it's fine